### ISSUE TITLE
When Reading Off the Price Of A Product In A New Order Using TalkBack It Isn't Read Back Correctly

### ISSUE THREAD

**Issue by JB184351** [woocommerce/woocommerce-android#7409]
**Describe the bug**
A clear and concise description of what the bug is. Please be as descriptive as possible; issues lacking detail, or for any other reason than to report a bug, may be closed without action.

**To Reproduce**

1. Go to the 'Order' tab
2. Scroll down to 'Create a New Note'
4. Select the Product
5. Add Product
6. Toggle TalkBack Feature
7. Select Product on New Order Page.
8. Notice how the dollar amount is read back when using TalkBack

**Screenshots**
<img src="https://user-images.githubusercontent.com/36495451/190729139-210244aa-fe50-4959-bd68-1d970c2e85b7.jpg" width="300"/>

**Expected behavior**
The dollar amount should always be read back correctly when reading back a product to the user.

**Isolating the problem (mark completed items with an [x]):**
- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: Samsung Galaxy A51
- Android version: 12
- WooCommerce Android version: 10.1
